api/src/legal-documents/scripts/convert-users-orga-cgu-data.js

```diff
@@ -0,0 +1,121 @@
+import { setTimeout } from 'node:timers/promises';
+
+import { DomainTransaction } from '../../../lib/infrastructure/DomainTransaction.js';
+import { Script } from '../../shared/application/scripts/script.js';
+import { ScriptRunner } from '../../shared/application/scripts/script-runner.js';
+import { LegalDocumentService } from '../domain/models/LegalDocumentService.js';
+import { LegalDocumentType } from '../domain/models/LegalDocumentType.js';
+import * as legalDocumentRepository from '../infrastructure/repositories/legal-document.repository.js';
+
+const { TOS } = LegalDocumentType.VALUES;
+const { PIX_ORGA } = LegalDocumentService.VALUES;
+
+const DEFAULT_BATCH_SIZE = 1000;
+const DEFAULT_THROTTLE_DELAY = 300;
+
+export class ConvertUsersOrgaCguData extends Script {
+  constructor() {
+    super({
+      description: 'Convert users orga CGU data to legal-document-versions-user-acceptances',
+      permanent: false,
+      options: {
+        dryRun: {
+          type: 'boolean',
+          describe: 'Execute the script in dry run mode',
+          default: false,
+        },
+        batchSize: {
+          type: 'number',
+          describe: 'Size of the batch to process',
+          default: DEFAULT_BATCH_SIZE,
+        },
+        throttleDelay: {
+          type: 'number',
+          describe: 'Delay between batches in milliseconds',
+          default: DEFAULT_THROTTLE_DELAY,
+        },
+      },
+    });
+  }
+
+  async handle({ options, logger }) {
+    const { dryRun, batchSize, throttleDelay } = options;
+
+    const legalDocument = await legalDocumentRepository.findLastVersionByTypeAndService({
+      type: TOS,
+      service: PIX_ORGA,
+    });
+
+    if (!legalDocument) {
+      throw new Error(`No legal document found for type: ${TOS}, service: ${PIX_ORGA}`);
+    }
+
+    let hasMoreUsers = true;
+    let offset = 0;
+    let migrationCount = 0;
+
+    while (hasMoreUsers) {
+      const usersWithCgu = await this.#findOrgaCguAcceptedUsers({ batchSize, offset });
+
+      if (usersWithCgu.length === 0) {
+        hasMoreUsers = false;
+        break;
+      }
```

Comment:
La variable `hasMoreUsers` n'est pas vraiment utile : 

```suggestion
    let offset = 0;
    let migrationCount = 0;

    while (true) {
      const usersWithCgu = await this.#findOrgaCguAcceptedUsers({ batchSize, offset });

      if (usersWithCgu.length === 0) {
        break;
      }
```

Reply:
`while(true)` est interdit par eslint 😅

Reply:
Bien reçu que `while(true)` est interdit par eslint. Et je suis d'accord que c'est plus propre de ne pas faire `while(true)`  :+1: 

Retravailler alors un peu le code pour ne pas utiliser le `break` puisqu'on veut plutôt favoriser l'utilisation de `hasMoreUsers`.

Reply:
En passant par une constante ça passe eslint.

Reply:
Cela revient au même, c'est contourner la bonne pratique mise en avant par ESLint. La bonne chose à faire est en fait vraiment d'utiliser la variable `hasMoreUsers`.

Reply:
Mais on peut arrêter de passer du temps sur ce point car c'est un script `permanent: false` :smile:

Reply:
Aussi je passe ce point en « resolved ».